engine/crates/composition/src/compose/enums.rs

```diff
@@ -3,23 +3,22 @@ use std::collections::HashSet;
 use super::*;
 use crate::subgraphs::{StringId, Subgraphs};
 
-pub(super) fn merge_enum_definitions(
-    first: &DefinitionWalker<'_>,
-    definitions: &[DefinitionWalker<'_>],
-    ctx: &mut Context<'_>,
+pub(super) fn merge_enum_definitions<'a>(
+    first: &DefinitionWalker<'a>,
+    definitions: &[DefinitionWalker<'a>],
+    ctx: &mut Context<'a>,
 ) {
     let enum_name = first.name().id;
-    let is_inaccessible = definitions
-        .iter()
-        .any(|definition| definition.directives().inaccessible());
+    let directive_containers = definitions.iter().map(|def| def.directives());
```

Comment:
Not related to this PR, but why is it named `DirectiveContainerId`, why not `DirectiveId`? Does the `container` notion for a directive mean something specific?

Reply:
I would use `DIrectiveLocationId` mapping to a list of directives from your description

Reply:
It's something I introduced so we can handle directives uniformly across different kinds of schema items. It really payed off in ease of implementation of the last few composition features.

Reply:
Renamed to DirectiveSite, even shorter.